core/src/alpenglow_consensus/vote_pool.rs

```diff
@@ -115,8 +115,14 @@ impl<VC: VoteCertificate> VotePool<VC> {
         Ok(())
     }
 
-    pub fn has_prev_vote(&self, validator_key: &Pubkey) -> bool {
-        self.prev_votes.contains_key(validator_key)
+    pub(crate) fn has_prev_vote(&self, validator_key: &Pubkey, vote_key: Option<&VoteKey>) -> bool {
```

Comment:
after we stabilize BLS we can refactor certificate pool to be slot based and hopefully clean up a lot of this. 


Also in favor of just removing the conflicting vote check and leaving it to slashing. Just respect the # of votes per type

Reply:
The conflicting vote check is added partly because I wanted to avoid double counting stake. If you send Notarize(b) and NotarizeFallback(b) I can't count your stake twice. Slashing doesn't happen real time, I don't think we can totally rely on slashing for everything.

Reply:
That's fine, we can discuss that when we refactor certificate pool. But I definitely want to keep the double counting stake check.